### PR TITLE
AGR-2319

### PR DESCRIPTION
**Pull request by ianlongden**
Removal of most of the hard coded urls.

resource_descriptor_helper no longer used and resource_descriptor_helper_2 used instead.
_2 was a better start point for me as it was easier to understand and dicts were used instead of lists which made it much faster for searching which there was a lot of.

If i edited a file then I will have made it flake8 compliant. (By hand so should be safe)

If there are any problems during the load wrt etl_helper then critical messages are displayed and there are now some
counts of how many times this happened kept. ETL now has a error_messages method that will give summarys of any errors that occurred (Otherwise the output can be too verbose).

resource_descriptor_helper_2 no longer forces an exit on a problem but reports it as above, it is up to the user to check for these critical error messages to ensure all is good. Some messages are fine, I was just being extremely careful if I failed to look up a url for some reason.

Originally I checked the urls that are returned by producing them from the old hardcoded method and the new one and if there were any differences they were reported (bad_pages dict). After passing several times through make test_reload and no differences reported or at least none that made a difference, the old hard coded method was removed.
So basically testing that the correct urls are generated was tested and afterwards testing removed for speed/sanity.

### CODE REVIEW

**Comment by ianlongden** [comment]
Tests will fail until agr_schema branch ZFIN_default_url is merged.

**Comment by christabone** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 71
- Added 91
           

Complexity increasing per file
==============================
- src/test/unit_tests.py  12
         

Complexity decreasing per file
==============================
+ src/etl/helpers/etl_helper.py  -9
+ src/etl/phenotype_etl.py  -1
         

Clones added
============
- src/etl/transcript_etl.py  1
- src/etl/stub_etl.py  1
- src/etl/disease_etl.py  1
- src/etl/closure_etl.py  1
- src/etl/vep_transcript_etl.py  1
- src/etl/vep_etl.py  1
- src/etl/variation_etl.py  2
- src/etl/construct_etl.py  1
- src/etl/expression_etl.py  1
- src/etl/generic_ontology_etl.py  1
- src/etl/phenotype_etl.py  4
         

Clones removed
==============
+ src/etl/go_etl.py  -1
+ src/etl/affected_genomic_model_etl.py  -3
+ src/etl/do_etl.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Alliance_of_Genome_Resources/agr_loader/pullRequest?prid=6040213&bid=19791696)

**Comment by christabone** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 71
- Added 91
           

Complexity increasing per file
==============================
- src/test/unit_tests.py  12
         

Complexity decreasing per file
==============================
+ src/etl/helpers/etl_helper.py  -9
+ src/etl/phenotype_etl.py  -1
         

Clones added
============
- src/etl/disease_etl.py  1
- src/etl/variation_etl.py  2
- src/etl/generic_ontology_etl.py  1
- src/etl/stub_etl.py  1
- src/etl/expression_etl.py  1
- src/etl/closure_etl.py  1
- src/etl/vep_etl.py  1
- src/etl/construct_etl.py  1
- src/etl/transcript_etl.py  1
- src/etl/vep_transcript_etl.py  1
- src/etl/phenotype_etl.py  4
         

Clones removed
==============
+ src/etl/do_etl.py  -1
+ src/etl/go_etl.py  -1
+ src/etl/affected_genomic_model_etl.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Alliance_of_Genome_Resources/agr_loader/pullRequest?prid=6040213&bid=19791696)

**Comment by christabone** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 73
- Added 70
           

Complexity increasing per file
==============================
- src/test/unit_tests.py  12
         

Complexity decreasing per file
==============================
+ src/etl/helpers/etl_helper.py  -9
+ src/etl/phenotype_etl.py  -1
         

Clones added
============
- src/etl/transcript_etl.py  1
- src/etl/vep_transcript_etl.py  1
- src/etl/variation_etl.py  2
- src/etl/stub_etl.py  1
- src/etl/generic_ontology_etl.py  1
- src/etl/disease_etl.py  1
- src/etl/closure_etl.py  1
- src/etl/expression_etl.py  1
- src/etl/construct_etl.py  1
- src/etl/phenotype_etl.py  4
- src/etl/vep_etl.py  1
         

Clones removed
==============
+ src/etl/go_etl.py  -1
+ src/etl/affected_genomic_model_etl.py  -3
+ src/etl/do_etl.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Alliance_of_Genome_Resources/agr_loader/pullRequest?prid=6040213&bid=19791696)

**Comment by christabone** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 72
- Added 75
           

Complexity increasing per file
==============================
- src/test/unit_tests.py  12
         

Complexity decreasing per file
==============================
+ src/etl/affected_genomic_model_etl.py  -20
+ src/etl/phenotype_etl.py  -1
+ src/etl/helpers/etl_helper.py  -9
         

Clones added
============
- src/etl/vep_transcript_etl.py  1
- src/etl/disease_etl.py  1
- src/etl/closure_etl.py  1
- src/etl/phenotype_etl.py  4
- src/etl/expression_etl.py  1
- src/etl/vep_etl.py  1
- src/etl/stub_etl.py  1
- src/etl/generic_ontology_etl.py  1
- src/etl/transcript_etl.py  1
- src/etl/construct_etl.py  1
- src/etl/variation_etl.py  2
         

Clones removed
==============
+ src/etl/affected_genomic_model_etl.py  -3
+ src/etl/go_etl.py  -1
+ src/etl/do_etl.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Alliance_of_Genome_Resources/agr_loader/pullRequest?prid=6040213&bid=19791696)

**Comment by ianlongden** [comment]
Closing for now until I have finished some extra stuff.
